src/org/labkey/test/components/glassLibrary/grids/DetailTableEdit.java

```diff
@@ -164,38 +165,24 @@ public boolean getBooleanField(String fieldCaption)
     public DetailTableEdit setBooleanField(String fieldCaption, boolean value)
     {
 
-        if(isFieldEditable(fieldCaption))
-        {
+        WebElement fieldValueElement = elementCache().fieldValue(fieldCaption).findElement(getComponentElement());
+        Assert.isTrue(isEditableField(fieldValueElement), String.format("Field '%s' is not editable, and cannot be set.", fieldCaption));
 
-            WebElement fieldValueElement = elementCache().fieldValue(fieldCaption).findElement(getComponentElement());
+        // The text used in the field caption and the value of the name attribute in the checkbox don't always have the same case.
+        WebElement editableElement = fieldValueElement.findElement(Locator.tagWithAttributeIgnoreCase("input", "name", fieldCaption));
+        String elementType = editableElement.getAttribute("type").toLowerCase().trim();
 
-            WebElement editableElement = fieldValueElement.findElement(Locator.tagWithName("input", fieldCaption.toLowerCase()));
-            String elementType = editableElement.getAttribute("type").toLowerCase().trim();
+        Assert.isTrue(elementType.equals("checkbox"), String.format("Field '%s' is not a checkbox, cannot be set to true/false.", fieldCaption));
 
-            if(elementType.equals("checkbox"))
-            {
-                Checkbox checkbox = new Checkbox(editableElement);
-
-                if(checkbox.isChecked() && !value)
-                {
-                    // If value is false and the checkbox is checked, uncheck it.
-                    checkbox.check();
-                }
-                else if(!checkbox.isChecked() && value)
-                {
-                    // If value is true and the checkbox is not checked, check it.
-                    checkbox.check();
-                }
+        Checkbox checkbox = new Checkbox(editableElement);
 
-            }
-            else
-            {
-                throw new NoSuchElementException("Field '" + fieldCaption + "' is not a checkbox, cannot be set to true/false.");
-            }
+        if(!value)
+        {
+            checkbox.uncheck();
         }
         else
         {
-            throw new IllegalArgumentException("Field with caption '" + fieldCaption + "' is read-only. This field can not be set.");
+            checkbox.check();
         }
```

Comment:
Checkbox has a `set` method:
```suggestion
        checkbox.set(value);
```

Reply:
Someday I will remember that.